apps/prairielearn/python/prairielearn.py

```diff
@@ -284,7 +289,7 @@ def to_json(v, *, df_encoding_version=1, np_encoding_version=1):
             "_value": str(v),
         }
 
-    if np.isscalar(v) and np.iscomplexobj(v):
+    if np.isscalar(v) and isinstance(v, np.complexfloating) and np.iscomplexobj(v):
```

Comment:
This `isinstance(np.complexfloating)` is similar to the function call inside the `iscomplexobj` code. Do we want to just keep that call? Otherwise, it is hard to narrow the type.

Reply:
I think it's fine to have just the np.iscomplexobj with an ignore comment. If we run into this issue more we can add a wrapper function or something.

Reply:
Changed in [1d5744a](https://github.com/PrairieLearn/PrairieLearn/pull/11192/commits/1d5744a80740b682c3723bd2bd1b87cdc2836204).